source/mesh_deformation/fastscapecc.cc

```diff
@@ -0,0 +1,698 @@
+#include <iostream>
+
+#include <xtensor/xarray.hpp>
+#include <xtensor/xmath.hpp>
+#include <xtensor/xview.hpp>
+#include <xtensor/xadapt.hpp>
+
+#include <fastscapelib/flow/flow_graph.hpp>
+#include <fastscapelib/flow/sink_resolver.hpp>
+#include <fastscapelib/flow/flow_router.hpp>
+#include <fastscapelib/grid/raster_grid.hpp>
+#include <fastscapelib/eroders/diffusion_adi.hpp>
+#include <fastscapelib/eroders/spl.hpp>
+
+
+#include <aspect/global.h>
+
+#include <aspect/mesh_deformation/fastscapecc.h>
+#include <aspect/geometry_model/box.h>
+#include <deal.II/numerics/vector_tools.h>
+#include <aspect/postprocess/visualization.h>
+#include <ctime>
+#include <aspect/simulator.h>
+#include <aspect/simulator/assemblers/interface.h>
+
+#include <algorithm> // For std::copy
+
+
+namespace aspect
+{
+  namespace MeshDeformation
+  {
+    template <int dim>
+    void
+    FastScapecc<dim>::initialize ()
+    {
+        AssertThrow(Plugins::plugin_type_matches<const GeometryModel::Box<dim>>(this->get_geometry_model()),
+                         ExcMessage("FastScape can only be run with a box geometry model."));
+
+             const GeometryModel::Box<dim> *geometry
+               = dynamic_cast<const GeometryModel::Box<dim>*> (&this->get_geometry_model());
```

Comment:
You will have to run `./contrib/utilities/indent` to get a consistent indentation of the entire file. You'll like the end result, it will be much easier to read!